src/pydvl/valuation/methods/classwise_shapley.py

```diff
@@ -211,23 +166,27 @@ def _normalize(self) -> ValuationResult:
         Returns:
             Normalized ValuationResult object.
         """
-        u = self.utility
+        if self.result is None:
+            raise ValueError("You should call fit before calling _normalize()")
 
-        assert self.result is not None
-        assert self.labels is not None
-        assert u.training_data is not None
+        if self.utility.training_data is None:
+            raise ValueError("You should call fit before calling _normalize()")
```

Comment:
There is also a `NotFittedException` in `pydvl.utils.exceptions`. The error message is a bit more generic though.

Reply:
I think that exception doesn't quite fit this error.

Reply:
Ok, but I would at least make it a RuntimeError, because it is not an error due to a wrong value a user provides.